### ISSUE TITLE
Improve notebook output cleanliness for published HTML documentation

### ISSUE THREAD

**Issue by jgallowa07** [matsengrp/SARS-CoV-2_spike_multidms#37]
# Improve notebook output cleanliness for published HTML documentation

## Summary

The exported HTML notebooks serve as the project's published documentation. Several output quality issues — blanket warning suppression, raw `pprint` config dumps, unexplained `.head()` calls, and inconsistent naming — reduce the professionalism and debuggability of these pages. This issue collects targeted fixes to make the HTML output cleaner.

## Background

All 20 notebooks are exported via `marimo export html` and served as GitHub Pages. They are the primary way readers interact with the analysis results. Current issues:

1. **Blanket warning suppression** — 12 notebooks use `warnings.simplefilter("ignore")`, hiding all warnings including potentially meaningful ones (e.g., convergence failures, deprecation warnings from JAX). This makes debugging silent failures very difficult.

2. **Raw `pprint` config dumps** — `spike_03` and `sim_02` use `pprint.pprint(fit_params)` to display the full fitting configuration dict. In the HTML output, this renders as an unformatted wall of text with no explanation.

3. **Unexplained `.head()` calls** — Several notebooks (`spike_04`, `spike_08`, `sim_04`, `sim_07`) print `df.head()` with no surrounding markdown explaining what the reader should observe. These look like debugging artifacts.

4. **`OUTDIR` alias** — `spike_structure_analysis.py` uses both `output_dir` (from config) and `OUTDIR` (hardcoded `"results/spike_structure_analysis/"`) for different purposes. The naming is confusing since both are output directories.

5. **Double groupby in `spike_02`** — The `is_stop` logic groups by `is_stop` twice in sequence with nearly identical operations.

### 1. Replace blanket warning suppression with targeted filters

**Files:** 12 notebooks listed above (existing)

Replace:
```python
warnings.simplefilter("ignore")
```

With targeted filters for known noisy warnings:
```python
warnings.filterwarnings("ignore", message=".*jax.*", category=DeprecationWarning)
warnings.filterwarnings("ignore", message=".*plotnine.*", category=FutureWarning)
```

The specific filters will depend on which warnings each notebook actually produces. Run each notebook and catalog the warnings before choosing filters.

### 2. Replace `pprint` config dumps with structured display

**Files:** `notebooks/spike/spike_03_fit_models.py`, `notebooks/simulation/sim_02_model_fitting.py` (existing)

Replace:
```python
pprint.pprint({k: v for k, v in fit_params.items() if k != "dataset"})
```

With a marimo `mo.md()` table or formatted markdown summary showing key parameters (model type, epochs, learning rate, fusionreg values, regularization) in a readable layout. Remove the `import pprint`.

### 3. Add context to `.head()` calls or remove them

**Files:** `spike_04`, `spike_08`, `sim_04`, `sim_07` (existing)

For each `.head()` call, either:
- Add a preceding `mo.md("### ...")` cell explaining what the reader should observe in the table, or
- Remove the `.head()` if it was purely a development artifact

### 4. Rename `OUTDIR` to `structure_output_dir`

**File:** `notebooks/supplemental/spike_structure_analysis.py` (existing)

Rename the local `OUTDIR = "results/spike_structure_analysis/"` to `structure_output_dir` (or similar) to distinguish it from `output_dir` which comes from config and points to `results/spike_analysis/`.

### 5. Simplify double groupby in `spike_02`

**File:** `notebooks/spike/spike_02_exploratory_stats.py` (existing)

Refactor the `is_stop` groupby logic to avoid the redundant second groupby. The exact refactor depends on what the two passes do differently, but the goal is a single pass.

## Testing Strategy

- Run full pipeline with test profile (`pixi run test`) to confirm all notebooks still execute without error
- Visually inspect the exported HTML for each modified notebook to verify:
  - No unexpected warnings leak through the targeted filters
  - Config display is readable and compact
  - `.head()` calls either have context or are removed
- Verify that `spike_structure_analysis` still produces `nbr_score_df.csv` after the rename

## Success Criteria

- [ ] No `warnings.simplefilter("ignore")` remains in any notebook
- [ ] `pprint` removed from `spike_03` and `sim_02`; config displayed in structured format
- [ ] Every `.head()` call either has explanatory markdown or is removed
- [ ] `OUTDIR` renamed in `spike_structure_analysis.py`
- [ ] `pixi run test` passes
